src/components/InteractiveQuestionView.tsx

```diff
@@ -0,0 +1,105 @@
+import { Box, Flex, Group, Modal } from "@mantine/core"
+import { InteractiveQuestion } from "@metabase/embedding-sdk-react"
+
+import { ThemedButton } from "./ThemedButton"
+import { CustomIcon } from "./CustomIcon"
+import { useReducer } from "react"
+import { useDisclosure } from "@mantine/hooks"
+
+// TODO: add the "breakout" view once we've separated the breakout from summarization in the SDK
+type QuestionView = "viz" | "filter" | "summary" | "editor"
+
+interface Props {
+  isSaveEnabled?: boolean
+}
+
+export const InteractiveQuestionView = (props: Props) => {
+  const { isSaveEnabled = false } = props
+
+  const [view, changeView] = useReducer(
+    (view: QuestionView, nextView: QuestionView) =>
+      view !== "viz" && view === nextView ? "viz" : nextView,
```

Comment:
This allows us to click on the same view's button again and go back to the visualization view.

Reply:
inlining this logic makes it a little hard to understand. What this does is essentially changing the view to `viz` if we're clicking the same button as the current view. Otherwise, change to whatever the nextViz is.

I'd like to see this being structured with some if blocks, that might make this more readable.

Reply:
@WiNloSt Good idea. I initially has this explicitly written with if-blocks, but later collapsed to this. I'll revert this then, as the former version is indeed more readable.